app/src/components/DataRequirements.tsx

```diff
@@ -0,0 +1,142 @@
+import { Accordion, Center, createStyles, em, getBreakpointValue, List, Paper, rem } from '@mantine/core';
+
+interface CodeInterface {
+  code: string | undefined;
+  system: string | undefined;
+}
+
+const useStyles = createStyles(theme => ({
+  card: {
+    borderRadius: 12,
+    border: `${rem(2)} solid ${theme.colors.gray[3]}`,
+    width: '1200px',
+    [`@media (max-width: ${em(getBreakpointValue(theme.breakpoints.lg) - 1)})`]: {
+      width: '100%'
+    }
+  }
+}));
+
+/**
+ * Component which displays all data requirements of a specified resource and displays them
+ * as resource cards that contain all required information
+ */
+function DataRequirements({ type, extension, dateFilter, codeFilter }: fhir4.DataRequirement) {
+  const { classes } = useStyles();
+  const codes: CodeInterface[] = [];
+
+  const codeFiltersWithVS = codeFilter?.filter(cf => {
+    return cf?.valueSet !== undefined;
+  });
+
+  codeFilter?.forEach(cf => {
+    cf?.code?.forEach(c => {
+      const code: CodeInterface = { code: c.code, system: c.system };
+      codes.push(code);
+    });
+  });
+
+  return (
+    <Center>
+      <Paper className={classes.card} shadow="sm" p="md">
+        <Accordion variant="contained" radius="lg" defaultValue="customization">
+          <Accordion.Item value="customization">
+            <Accordion.Control>
+              <Center>
+                <h2>{type} </h2>
+              </Center>
+            </Accordion.Control>
+            <Accordion.Panel>
+              <List>
+                <List.Item>
+                  <b>Extension(s): </b>
+                  <br />
+                </List.Item>
+                <List withPadding>
+                  {extension?.map(e => (
+                    <>
+                      <List.Item key={e.url}>
+                        <b>URL:</b>
+                        {e.url}
+                      </List.Item>
+                      {e.valueString && (
+                        <List.Item key={e.valueString}>
+                          <b> ValueString: </b>
+                          {e.valueString}
+                        </List.Item>
+                      )}
+                      <br />
+                    </>
+                  ))}
+                </List>
+                {dateFilter && dateFilter?.length > 0 && (
+                  <>
+                    <List.Item>
+                      <b>DateFilter(s):</b>
+                    </List.Item>
+                    <List withPadding>
+                      {dateFilter.map(date => (
+                        <>
+                          <List.Item key={date?.valuePeriod?.start}>
+                            <b> Start Date: </b>
+                            {date?.valuePeriod?.start} <br />
+                          </List.Item>
+                          <List.Item key={date?.valuePeriod?.end}>
+                            <b> End Date: </b> {date?.valuePeriod?.end} <br />
```

Comment:
```suggestion
                          <List.Item key={date.valuePeriod?.start}>
                            <b> Start Date: </b>
                            {date.valuePeriod?.start} <br />
                          </List.Item>
                          <List.Item key={date.valuePeriod?.end}>
                            <b> End Date: </b> {date.valuePeriod?.end} <br />
```
I don't think we need these since we are doing a `.map` on the dateFilter array so we know each entry is defined as a `DataRequirementDateFilter`